sway-lib-std/src/hash.sw

```diff
@@ -124,3 +124,23 @@ pub fn hash_pair(value_a: b256, value_b: b256, method: HashMethod) -> b256 {
         }
     }
 }
+
+/// Returns the SHA-2-256 hash of `param`.
+pub fn sha256<T>(param: T) -> b256 {
+    let result_buffer: b256 = 0x0000000000000000000000000000000000000000000000000000000000000000;
```

Comment:
Hmmmm. This might not be the most important, but would making `result_buffer` `mut` be more semantically correct? The compiler can't really analyze what's in asm blocks yet, but it would just be more info to both the reader and potentially later the compiler.

Also, can you use `ZERO` instead of the literal `0x0000000000000000000000000000000000000000000000000000000000000000`? Not sure if it actually works. https://github.com/FuelLabs/sway/blob/afcc5211d3e6bbcebecdb1b977ae4a44dc7e9c2b/sway-lib-std/src/constants.sw#L4

Reply:
The `mut` does make more sense to me but then I doubt that `const` would work if I tried to mix them both so it's probably one or the other

Reply:
Agree we should be using `mut` here even if it doesn't seem to make a difference now.

Reply:
Ah I meant

```rust
let mut result_buffer: b256 = ZERO;
```

which should work just as well as using the literal `0x0000000000000000000000000000000000000000000000000000000000000000`.